code/modules/antagonists/heretic/items/eldritch_painting.dm

```diff
@@ -0,0 +1,448 @@
+// The basic eldritch painting
+/obj/item/wallframe/painting/eldritch
+	name = "The debug and a coder who slept"
+	desc = "This shouldn't be shown..."
+	icon = 'icons/obj/signs.dmi'
+	resistance_flags = FLAMMABLE
+	flags_1 = NONE
+	icon_state = "frame-empty"
+	result_path = /obj/structure/sign/painting/eldritch
+	pixel_shift = 30
+
+/obj/structure/sign/painting/eldritch
+	name = "The debug and a coder who slept"
+	desc = "This shouldn't be shown..."
+	icon = 'icons/obj/signs.dmi'
+	icon_state = "frame-empty"
+	custom_materials = list(/datum/material/wood =SHEET_MATERIAL_AMOUNT)
+	resistance_flags = FLAMMABLE
+	buildable_sign = FALSE
+	// The list of canvas types accepted by this frame, set to zero here
+	accepted_canvas_types = list()
+	// A basic proximity sensor
+	var/datum/proximity_monitor/painting_proximity_sensor
+	// For changing sensor types on subtypes
+	var/sensor_type = /datum/proximity_monitor/advanced/eldritch_painting
+	// Set to false since we don't want this to persist
+	persistence_id = FALSE
+
+// Mood applied for ripping the painting
+/datum/mood_event/eldritch_painting
+	description = "YOU, I SHOULD NOT HAVE DONE THAT!!!"
+	mood_change = -6
+	timeout = 3 MINUTES
+
+/obj/structure/sign/painting/eldritch/Initialize(mapload, dir, building)
+	if(sensor_type)
+		painting_proximity_sensor = new sensor_type(_host = src, range = 7, _ignore_if_not_on_turf = TRUE)
+	return ..()
+
+/obj/structure/sign/painting/eldritch/wirecutter_act(mob/living/user, obj/item/I)
+	if(!user.can_block_magic(MAGIC_RESISTANCE))
+		user.add_mood_event("ripped_eldritch_painting", /datum/mood_event/eldritch_painting)
+		to_chat(user, span_notice("Laughter echoes through your mind...."))
+	QDEL_NULL(painting_proximity_sensor)
+	qdel(src)
+
+// Applies an affect on view
+/datum/proximity_monitor/advanced/eldritch_painting
```

Comment:
you should have this proximity monitor in its own file in the proximity monitor folder, not here

Reply:
Fixed in latest commit